### DISCUSSION TITLE
Cycle between windows of the same app?

### DISCUSSION THREAD

**Discussion by AlgusDark** [asmvik/yabai#1326] [Q&A] [answered]
I'm looking into creating the following thing:

- one key launch or focus app (hyper+b opens/or focus brave)
  - [x] skdh >> `cmd + ctrl + alt - b : open -a "Brave Browser"`
- leader key (?) For opening new windows (Ctrl+shift+n [we can release it and then press] b h> opens new brave window)
  - [x] skhd modes >> `open_new < b : open -na "Brave Browser" ; skhd -k "escape"`
- For the first one-key-launcher, I would like to cycle between apps in across multiple spaces/screens if my window is focused in the app and using the same one-key-launcher.

I believe that could be done with a helper like this:
- Getting the current opening apps by name:
  - `yabai -m query --windows | jq 'map(select(.app=="Brave Browser").id) | .[(index(ANY_ID)+1)%length] '`
- And for the third thing... how can I get an order of the query I posted?
  - Maybe I don't need the order but to get the id of the current app and cycle with my previous query 🤔 

Is that possible? Some code would be appreciated 🙏

**Comment by Bellavene** [comment, marked as answer]
Just make a script for native macos shortcut  cmd + ` and add it after open -a "Brave Browser";

**Reply by AlgusDark** [reply]
I forgot that I had this post. Thanks for bringing this up and making a suggestion 🙏

I solved it with a custom script like this:

```sh
#!/bin/zsh

while [[ "$#" -gt 0 ]]
  do
    case $1 in
      -f|--focus)
        if [ "$2" = "prev" ]
        then
          pos=-1
        else
          pos=1
        fi
        ;;
    esac
    shift
  done

pos=${pos:-1}

focus="$(yabai -m query --windows | \
    jq -e --argjson pos $pos '(.[] | select(."has-focus")) as {$id, $app}
      | map( select( .app==$app and ((."is-hidden" or ."is-minimized") | not) ) )
      | sort_by(.space, .frame.x, .frame.y)
      | map(.id)
      | .[(index($id)+($pos))%length]'
  )"

yabai -m window --focus "$focus"
```

**Reply by Bellavene** [reply]
Nice one
